niap-cc/Permissions/Companion/app/src/main/java/com/android/certifications/niap/permissions/companion/MainActivity.java

```diff
@@ -83,33 +93,108 @@ public class MainActivity extends AppCompatActivity {
             Manifest.permission.ACCESS_FINE_LOCATION,
     };
 
-    private TextView mStatusTextView;
+    private ListView mStatusListView;
+    static List<String> mStatusData = new ArrayList<String>();
+    static ArrayAdapter<String> mStatusAdapter;
+
+    protected void setAdapters(){
+        mStatusAdapter = new ArrayAdapter<String>(
+                this,
+                android.R.layout.simple_list_item_1,
+                mStatusData);
+        mStatusListView.setAdapter(mStatusAdapter);
+    }
     private Button mSetupButton;
+    private CheckBox mCheckBoxButton;
     private boolean mGmsAvailable;
 
     /**
      * Used to ensure the GMS location settings are configured as required for the location tests.
      */
     private LocationRequest mLocationRequest;
+    private void logdebug(String text)
+    {
+        Log.d(TAG, text);
+        runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if(mStatusAdapter != null){
+                    mStatusAdapter.add("🟩"+text);
+                }
+            }
+        });
+    }
+    private void logerror(String text)
+    {
+        Log.e(TAG, text);
+        runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if(mStatusAdapter != null){
+                    mStatusAdapter.add("🔴"+text);
+
+                }
+            }
+        });
+
+    }
+    public boolean deviceConfigSetProperty(String a,String b,String value,boolean makeDefault) {
```

Comment:
Ultra-nit: In a follow-up pull request, can you rename the parameters a and b to something more descriptive? From looking at the `setProperty` method, I think they should be `namespace` and `name` respectively, but I wasn't sure from the reflective call.

Reply:
Thanks! Will update it.

Reply:
Sorry I forgot to update the companion apps. updated. thanks.